ecosystem/gstreamer_plugin/gst_mtl_st40p_tx.c

```diff
@@ -324,6 +324,18 @@ static gboolean gst_mtl_st40p_tx_session_create(Gst_Mtl_St40p_Tx* sink) {
   }
   ops_tx.port.payload_type = sink->portArgs.payload_type;
 
+  if (sink->did > 0xFF) {
+    GST_ERROR("Invalid DID value: %d", sink->did);
+    return FALSE;
+  }
+  if (sink->sdid > 0xFF) {
+    GST_ERROR("Invalid SDID value: %d", sink->sdid);
+    return FALSE;
+  }
+  if (sink->framerate == 0) {
+    GST_INFO("Framerate not set, defaulting to 60");
+    sink->framerate = 60;
+  }
```

Comment:
Any way we can't take that from caps ? 
What if the caps negotiation will say we have 30 fps ? 
Here the fps variable is a must where we could just take it from values passed by the capabilities ?

Reply:
This is a bit tricky as there is no negotiation of capabilities in st40-tx as we can send any raw data in payload I don't know of any plugin that can provide any meaningful capabilities. I wrote it in such a way that all arguments are taken from the parameters to the plugin.